builtin/assign_osh.py

```diff
@@ -203,28 +204,85 @@ def _AssignVarForBuiltin(mem, rval, pair, which_scopes, flags):
     Used for assignment builtins, (( a = b )), {fd}>out, ${x=}, etc.
     """
     lval = LeftName(pair.var_name, pair.blame_word)
-    if pair.plus_eq:
+
+    initializer = None  # type: value.InitializerList
+    if rval is None:
+        # When 'export e+=', then rval is value.Str('')
+        # When 'export foo', the pair.plus_eq flag is false.
+        # Thus, when rval is None, plus_eq cannot be True.
+        assert not pair.plus_eq, pair.plus_eq
+        # NOTE: when rval is None, only flags are changed
+        val = None  # type: value_t
+    elif rval.tag() == value_e.InitializerList:
+        old_val = sh_expr_eval.OldValue(
+            lval,
+            mem,
+            None,  # ignore set -u
+            pair.blame_word)
+        initializer = cast(value.InitializerList, rval)
+
+        # OSH compatibility: "declare -A assoc=(1 2)" is a usage
+        # error.  This code will soon be removed when the
+        # initialization of the form "assoc=(key value)" is supported.
+        if flag_A and len(initializer.assigns
```

Comment:
I would add a () around the entire `if` condition and then reformat with yapf -- I do that because that way yapf can give a more natural wrapping

Reply:
I fixed it in 28bf152247.